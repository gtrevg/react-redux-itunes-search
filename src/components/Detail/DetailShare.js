import React from 'react';

const DetailShare = () => {
	return <div>Share</div>;
};

export default DetailShare;
