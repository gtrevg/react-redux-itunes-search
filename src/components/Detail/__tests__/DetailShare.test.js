import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';

import DetailShare from '../DetailShare';

describe('DetailShare component', () => {
	const component = (
		<DetailShare
			title="Billie Jean"
			artist="Michael Jackson"
			url="https://itunes.apple.com/us/album/billie-jean/269572838?i=269573364&uo=4"
		/>
	);

	it('Renders without crashing', () => {
		const div = document.createElement('div');
		ReactDOM.render(component, div);
		ReactDOM.unmountComponentAtNode(div);
	});

	it('Snapshot matchs', () => {
		const tree = renderer.create(component).toJSON();
		expect(tree).toMatchSnapshot();
	});
});
