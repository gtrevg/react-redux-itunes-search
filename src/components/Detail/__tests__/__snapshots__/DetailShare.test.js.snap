// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DetailShare component Snapshot matchs 1`] = `
<div>
  <mocked
    name="React-Share FacebookShareButton"
    props={
      Object {
        "className": "DetailShareButton",
        "quote": "Listening Billie Jean by Michael Jackson",
        "url": "https://itunes.apple.com/us/album/billie-jean/269572838?i=269573364&uo=4",
      }
    }
  >
    <mocked
      name="React-Share FacebookIcon"
      props={
        Object {
          "round": true,
          "size": 48,
        }
      }
    />
  </mocked>
  <mocked
    name="React-Share GooglePlusShareButton"
    props={
      Object {
        "className": "DetailShareButton",
        "url": "https://itunes.apple.com/us/album/billie-jean/269572838?i=269573364&uo=4",
      }
    }
  >
    <mocked
      name="React-Share GooglePlusIcon"
      props={
        Object {
          "round": true,
          "size": 48,
        }
      }
    />
  </mocked>
  <mocked
    name="React-Share TwitterShareButton"
    props={
      Object {
        "className": "DetailShareButton",
        "title": "Listening Billie Jean by Michael Jackson",
        "url": "https://itunes.apple.com/us/album/billie-jean/269572838?i=269573364&uo=4",
      }
    }
  >
    <mocked
      name="React-Share TwitterIcon"
      props={
        Object {
          "round": true,
          "size": 48,
        }
      }
    />
  </mocked>
</div>
`;
